src/PresentationalComponents/ComplianceTable/ComplianceTable.js

```diff
@@ -31,13 +31,13 @@ const ComplianceTable = (props) => {
   const tableState = useRawTableState();
 
   useDeepCompareEffectNoCheck(() => {
-    if (ENABLE_ASYNC_TABLE_HOOKS) {
+    if (useAPIV2FeatureFlag) {
```

Comment:
We have to call the hook and use the returned value as a feature flag value.

```suggestion
   const apiV2Enabled = useAPIV2FeatureFalg();
   ...
   
    if (apiV2Enabled) {
```